api/src/main.rs

```diff
@@ -17,25 +17,15 @@ async fn main() {
     env_logger::init();
     log::info!("Starting service");
 
-    let configuration = {
-        dotenv().ok();
-        match Configuration::init_from_env() {
-            Ok(val) => val,
-            Err(e) => {
-                log::error!("{}", e);
-                log::error!("Terminating because of previous error.");
-                process::exit(1);
-            }
-        }
-    };
-    match CONFIGURATION.set(configuration) {
-        Ok(()) => log::info!("Successfully provided global service configuration"),
-        Err(_) => {
-            log::error!("Failed to provide global service configuration");
-            log::error!("Terminating because of previous error.");
-            process::exit(1);
-        }
-    }
+    dotenv().ok();
+    let conf = Configuration::init_from_env()
+        .map_err(log_error_and_panic)
```

Comment:
Shouldn't this be [`.or_else(...)`](https://doc.rust-lang.org/std/result/enum.Result.html#method.or_else)?
Because `.map_err` will run even though it's `Ok(...)`.

Reply:
From the documentation:
```
Maps a Result<T, E> to Result<T, F> by applying a function to a contained Err value, leaving an Ok value untouched.
This function can be used to pass through a successful result while handling an error.
```
The `.or_else()` should work as well and would be required if we are not terminating the program.